agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -434,14 +439,12 @@ impl DeriveSchema for Stage {
             project: &ProjectStage,
             state: &mut ResultSetState,
         ) -> Result<Schema> {
+            state.result_set_schema = promote_missing(&state.result_set_schema);
             // If this is an exclusion $project, we can remove the fields from the schema and
             // return
-            state.result_set_schema = promote_missing(&state.result_set_schema);
-            if project
-                .items
-                .iter()
-                .all(|(k, p)| matches!(p, ProjectItem::Exclusion) || k == "_id")
-            {
+            if project.items.iter().all(|(k, p)| {
+                matches!(p, ProjectItem::Exclusion) || (k == "_id" && project.items.len() > 1)
```

Comment:
This is needed when you use _id as an alias and its the only key, as in the "_id is a valid alias" spec test.